src/core/src/op/cum_sum.cpp

```diff
@@ -58,8 +59,89 @@ shared_ptr<Node> op::v0::CumSum::clone_with_new_inputs(const OutputVector& new_a
     }
 }
 
-NGRAPH_SUPPRESS_DEPRECATED_START
+OPENVINO_SUPPRESS_DEPRECATED_START
 shared_ptr<Node> op::v0::CumSum::get_default_value() const {
     return ngraph::make_constant_from_string("0", get_element_type(), get_shape());
 }
-NGRAPH_SUPPRESS_DEPRECATED_END
+OPENVINO_SUPPRESS_DEPRECATED_END
+
+namespace {
+template <element::Type_t DATA_ET, element::Type_t AXIS_ET>
+bool evaluate_cum_sum(const HostTensorPtr& output,
+                      const HostTensorPtr& data,
+                      const HostTensorPtr& axis,
+                      const bool exclusive,
+                      const bool reverse) {
+    using data_t = fundamental_type_for<DATA_ET>;
+    using axis_t = fundamental_type_for<AXIS_ET>;
+    ngraph::runtime::reference::cumsum<data_t, axis_t>(data->get_data_ptr<data_t>(),
+                                                       axis->get_data_ptr<axis_t>(),
+                                                       output->get_data_ptr<data_t>(),
+                                                       data->get_shape(),
+                                                       exclusive,
+                                                       reverse);
+    return true;
+}
+
+#define CUM_SUM_TYPE_CASE(a, ...)                                          \
+    case element::Type_t::a: {                                             \
+        OV_OP_SCOPE(OV_PP_CAT4(evaluate_cum_sum, _, a, AXIS_ET));          \
+        return evaluate_cum_sum<element::Type_t::a, AXIS_ET>(__VA_ARGS__); \
+    }
+
+template <element::Type_t AXIS_ET>
+bool evaluate(const HostTensorPtr& output,
+              const HostTensorPtr& data,
+              const HostTensorPtr& axis,
+              const bool exclusive,
+              const bool reverse) {
+    switch (data->get_element_type()) {
+        CUM_SUM_TYPE_CASE(bf16, output, data, axis, exclusive, reverse);
```

Comment:
Dp we really need to have all these data types?

Reply:
Probably not. My aim was to cover the case previously supported by Template plugin. It's also not defined in the ticket which types should be handled here. @ilyachur would you advise?

Reply:
Done. Only float32 is considered.